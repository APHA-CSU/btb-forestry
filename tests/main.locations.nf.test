nextflow_process {

    name "Test Process locations"
    script "main.nf"
    process "locations"

    test("Should format location data") {

        when {
            params {
                today = "01Jan25"
            }
            process {
                """
                input[0] = file("${projectDir}/tests/data/location.csv")
                input[1] = file("${projectDir}/tests/data/counties.tsv")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
