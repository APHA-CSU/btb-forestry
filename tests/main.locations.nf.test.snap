{
    "Should format location data": {
        "content": [
            {
                "0": [
                    "allLocations_01Jan25.tsv:md5,de85a65c63778ded0ec223810c3223bf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-22T13:04:03.310153395"
    }
}